drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -586,7 +596,10 @@ local function initialize_switch(driver, device)
       battery_support = true
     end
     if #button_eps > 1 and tbl_contains(STATIC_BUTTON_PROFILE_SUPPORTED, #button_eps) then
-      if battery_support then
+      if device:supports_capability(capabilities.temperatureMeasurement) then
+        -- to call configure_buttons, keep the profile_name as nil.
+        device.log.debug("So far, it means Aqara Climate Sensor W100.")
+      elseif battery_support then
         profile_name = string.format("%d-button-battery", #button_eps)
       else
         profile_name = string.format("%d-button", #button_eps)
```

Comment:
```suggestion
      if device:supports_capability(capabilities.temperatureMeasurement) and device:supports_capability(capabilities.relativeHumidityMeasurement) then
        profile_name = "temperature-humidity"
      end
      if battery_support then
        profile_name = string.format("%d-button-battery", #button_eps) .. profile_name
      else
        profile_name = string.format("%d-button", #button_eps) .. profile_name
```

@nickolas-deboom do you think we should support full dynamicism here (a device:get_endpoints() call) or is leaving this as a fingerprint specific handling ok?

Reply:
the `.. profile_name` assumes the profile name has been changed according to my previous comment

Reply:
I am fine with it either way for now, especially since we'd just be re-profiling to the same profile it has already joined. If we start supporting more devices like this in the future then I'd definitely support making it more dynamic.

Reply:
@nickolas-deboom @hcarter-775 If we use fingerprint of `Aqara Climate Sensor W100` device, there is a problem that `configure_buttons(device)` is not called because a value is assigned to `profile_name`. So I modified to use the number of `temperature_eps` and `humidity_eps` obtained using `device:get_endpoints` to specify profile in the more dynamic way you mentioned.

```lua
    if #button_eps > 1 and tbl_contains(STATIC_BUTTON_PROFILE_SUPPORTED, #button_eps) then
      if #temperature_eps > 0 and #humidity_eps > 0 then
        device.log.debug("So far, it means Aqara Climate Sensor W100.")
        profile_name = "-temperature-humidity"
      end
      if battery_support then
        profile_name = string.format("%d-button-battery", #button_eps) .. profile_name
      else
        profile_name = string.format("%d-button", #button_eps) .. profile_name
      end
    elseif not battery_support then
      -- a battery-less button/remote
      profile_name = "button"
    end

    if profile_name ~= "" then
      device:try_update_metadata({profile = profile_name})
      device:set_field(DEFERRED_CONFIGURE, true)
      device:set_field(BUTTON_DEVICE_PROFILED, true)
    else
      configure_buttons(device)
    end
```